Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -387,9 +385,37 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
     
     // MARK: - Setting-up methods
     
+    /**
+     Updates key settings before loading the view.
+     
+     This method basically re-runs the setup which takes place in `init`. It could be useful if some of the attribtues have changed before `NavigationViewController` did load it's view, or if you did not have access to initializing logic. For example, as a part of `UIStoryboardSegue` configuration.
+     
+     - parameter routeResponse: `RouteResponse` object, containing selection of routes to follow.
+     - parameter routeIndex: The index of the route within the original `RouteResponse` object.
+     - parameter routeOptions: The route options used to get the route.
+     - parameter navigationOptions: The navigation options to use for the navigation session.
+     - returns `True` if setup was successful, `False` if `view` is already loaded and settings did not apply.
+     */
+    public func prepareViewLoading(routeResponse: RouteResponse, routeIndex: Int, routeOptions: RouteOptions, navigationOptions: NavigationOptions? = nil) -> Bool {
+        guard !isViewLoaded else {
```

Comment:
Why do we need to perform check to see if view was really loaded? I recall that in previous implementation in https://github.com/mapbox/mapbox-navigation-ios/pull/2974 it wasn't needed. Just want to make sure that this will not cause any side effects.

Reply:
This check is added since updating `routeResponse`, `routeIndex` or `routeOptions` will not have effect on the contents `NavigationViewController` if view is already loaded. In this case `view` - is the Navigation View, and these properties are only stored in `NavigationViewController` to initialize the NavigationView. So, this check lets user know if this update actually happened or not. Previous implementation did silent update of a local property, but that did not affect the NavigationView if it was already loaded.